### PR TITLE
Add schema in all class except operator

### PR DESCRIPTION
**Pull request by Quasimewdo**
Fix further errors when the table is in a non-default schema, such as
- df.apply(lambda t: ...)
- df[["id"]]

As result, this patch does modifications listing below
- Add schema in `DataFrame`, `func`, and `Type`
- Each class' qualified_name return `Tuple[str, str]` which corresponds
   to `schema` and `name`

This patch has also removed `operator`.

### CODE REVIEW

**Comment by Quasimewdo** [comment]
> Better to avoid name duplication with #131 so that we can search for the commit by name later easily.

Fix further error when table in non-default schema

File: greenplumpython/dataframe.py [hunk 1]
```diff
@@ -874,7 +886,8 @@ def _fetch(self, is_all: bool = True) -> Iterable[Tuple[Any]]:
 
     def save_as(
         self,
-        table_name: str,
+        dataframe_name: str,
+        schema_name: str = None,
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/dataframe.py:890]: Ok, let's keep it short

File: greenplumpython/dataframe.py [hunk 2]
```diff
@@ -679,7 +681,17 @@ def name(self) -> str:
         Returns:
             str: :class:`~dataframe.DataFrame`'s name.
         """
-        return self._name
+        return self._name if self._schema is None else f"{self._schema}.{self._name}"
+
+    @property
+    def _serialize(self) -> str:
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/dataframe.py]: I feel like qualified_name as "schema"."name" is easier to use than tuple[str, str]

File: tests/test_op.py [hunk 3]
```diff
@@ -39,42 +41,17 @@ def __init__(self, name: str, grade: int, courses: List[str]) -> None:
...
-        has_results=False,
-    )
-    complex = gp.type_("complex_number", schema="test")
-    complex_op = gp.operator("+", db=db, schema="test")
-    result = db.assign(complex_add=lambda: (complex_op(complex("(1, 2)"), complex("(1, 2)"))))
-    for row in result:
-        assert row["complex_add"] == {"i": 4, "r": 2}
-    complex_add = gp.function("complex_add", schema="test")
-    result = db.apply(lambda: complex_add(complex("(1, 2)"), complex("(1, 2)")), expand=True)
+    my_add = gp.operator("+", db=db)
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply tests/test_op.py:44]: We can put `gp.operator("+", db)`, but we can't remove `db`

File: greenplumpython/dataframe.py [hunk 4]
```diff
@@ -684,6 +688,16 @@ def name(self) -> str:
         """
         return self._name
 
+    @property
+    def qualified_name(self) -> Tuple[str, str]:
```
Comment (yihong0618) [inline greenplumpython/dataframe.py:692]: qualified_name -> qualified_name_tuple maybe better?

File: greenplumpython/type.py [hunk 5]
```diff
@@ -44,10 +45,23 @@ def __init__(
...
 
     def _serialize(self) -> str:
         obj_str = _serialize(self._obj)
-        return f"({obj_str}::{self._type_name})"
+        schema, type_name = self.qualified_name
+        qualified_name = f'"{schema}"."{type_name}"' if schema is not None else f'"{type_name}"'
+        return f"({obj_str}::{qualified_name})"
+
+    @property
+    def qualified_name(self) -> Tuple[str, str]:
```
Comment (yihong0618) [inline greenplumpython/type.py:57]: ditto

File: greenplumpython/type.py [hunk 6]
```diff
@@ -109,43 +126,54 @@ def __call__(self, obj: Any) -> TypeCast:
...
-        return TypeCast(obj, self._name)
+        return TypeCast(obj, self._name, self._schema)
 
     @property
     def name(self) -> str:
         """Get the name of this :class:`Type`."""
         return self._name
 
+    @property
+    def qualified_name(self) -> Tuple[str, str]:
```
Comment (yihong0618) [inline greenplumpython/type.py:144]: ditto

**Review by yihong0618 — APPROVED** [review]

File: tests/test_dataframe.py [hunk 7]
```diff
@@ -322,10 +322,18 @@ def test_table_distinct(db: gp.Database):
 
 
 def test_table_non_default_schema(db: gp.Database):
-    pg_class = db.create_dataframe(table_name="pg_catalog.pg_class")
+    pg_class = db.create_dataframe(table_name="pg_class", schema="pg_catalog")
```
Comment (beeender) [inline tests/test_dataframe.py:325]:
```
gpadmin=# create table "\a""" (id int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize sk
ew.
CREATE TABLE

df = db.create_dataframe(table_name="\\a\"")
```

Special chars in the table/schema name will still be a problem. But let's solve it in the future.

**Review by beeender — APPROVED** [review]

File: greenplumpython/col.py [hunk 8]
```diff
@@ -54,10 +54,10 @@ def __init__(self, name: str, dataframe: "DataFrame") -> None:
     def _serialize(self) -> str:
         assert self.dataframe is not None
         # Quote both dataframe name and column name to avoid SQL injection.
+        schema, df_name = self.dataframe.qualified_name
+        df_qualified_name = f'"{schema}"."{df_name}"' if schema is not None else f'"{df_name}"'
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/col.py:58]: Since all our test pass, which means this condition is not always true.

**Pull request merged**